src/day.rs

```diff
@@ -0,0 +1,231 @@
+use core::cmp::Ordering;
+use core::convert::TryFrom;
+use core::fmt;
+use core::fmt::Debug;
+use core::fmt::Display;
+use core::fmt::Write;
+use core::ops::Add;
+use core::ops::Sub;
+
+use crate::oldtime;
+use crate::DateTime;
+use crate::Days;
+use crate::NaiveDate;
+use crate::NaiveTime;
+use crate::TimeZone;
+use oldtime::Duration;
+
+/// Represents the full range of local timestamps in the day
+///
+///
+#[derive(Clone, Copy)]
+pub struct Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    date: NaiveDate,
+    tz: Tz,
+}
+
+impl<Tz> Debug for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        Debug::fmt(&self.date, f)?;
+        f.write_char(' ')?;
+        self.tz.fmt(f)
+    }
+}
+
+impl<Tz> Display for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+impl<Tz> PartialOrd for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.date.partial_cmp(&other.date)
+    }
+}
+
+impl<Tz> Ord for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.date.cmp(&other.date)
+    }
+}
+
+impl<Tz> PartialEq for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.date.eq(&other.date)
+    }
+}
+
+impl<Tz> Eq for Day<Tz> where Tz: TimeZone + Copy + Display {}
+
+impl<Tz> Day<Tz>
```

Comment:
Style nit: would prefer to have the inherent impl before trait impls.

Also I would write `impl<Tz: TimeZone + Copy + Display>` instead of having the `where` clause`, since it's a bit more concise.